src/main/java/dk/sdu/se_f22/productmodule/management/ProductManager.java

```diff
@@ -0,0 +1,356 @@
+                pT = p;
+                success = true;
+                break;
+            }
+        }
+
+        updateSource();
+
+        return success;
+    }
+
+    @Override
+    public boolean remove(String productId) {
+
+        //Removes a product from the array, and consequently from the source file as well.
+
+        checkForUpdates();
+        boolean toReturn = false;
+
+
+        for(Product p : productArray){
+            if (p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                toReturn = productArray.remove(p);
+                break;
+            }
+        }
+
+        updateSource();
+
+        return toReturn;
+    }
+
+    public boolean removeAll(String[] productIds){
+
+        //This function removes all products from the productArray that matches any of the given productIds
+        //It returns whether it was able to find products matching all the id's or not.
+
+        checkForUpdates();
+        int counter = 0;
+
+        for (String productId : productIds) {
+            for (Product p : productArray) {
+
+                if (p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)) {
+                    productArray.remove(p);
+                    counter++;
+                    break;
+                }
+            }
+        }
+
+        return counter == productIds.length;
+    }
+
+    @Override
+    public void setUpdateInterval(int time) {
+        checkForUpdates();
+
+        updateInterval = time;
+    }
+
+    private boolean backgroundUpdate(){
+
+        //This is a background function to be called once every index update interval.
+        //For direct control of index reparses, use reparse()
+
+        if(!backgroundThreadIsRunning) {
+            backgroundThreadIsRunning = true;
+        }
+
+        //Right here is where the XXXX.updateIndex() call to the module from Group 2.2 goes
+
+        return backgroundThread.isAlive();
+    }
+
+    @Override
+    public void run(){
+
+        //This function is only used by the backgroundThread.
+        //It busy-waits (Thread.sleep would work as well) for updateInterval (minutes)
+        //Then reads the json file again and prepares a new ArrayList<Product> for use.
+
+        while(System.currentTimeMillis() < lastCall + (updateInterval * 60000L)){
+            System.out.println();
+        }
+
+        lastCall = System.currentTimeMillis();
+        reparse();
+        backgroundUpdate();
+    }
+
+    @Override
+    public void reparse(){
+
+        //Reads a new productArray from the jsonReader.
+        //The current array is swapped out with the new one next time any CRUD operation is called.
+        //This is done as such, to prevent the backgroundThread and external calls to reparse() to cause issues.
+
+        try {
+            updatedProductArray = jsonReader.read();
+        }catch (IOException e){
+            e.printStackTrace();
+        }
+    }
+
+    private boolean checkForUpdates(){
```

Comment:
When it reparses, shouldn't it also update the actual productArray? 
Currently it only updates after any `crud operations` or `setUpdateInterval` has happened.

Reply:
We check for an updated product array before every crud operation so it always gets the newest. There should be no need to make sure the previous update is in memory as it is already on disk